src/helpers/predict_methods.py

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+from onnxruntime import InferenceSession
+from jaqpotpy.datasets.molecular_datasets import JaqpotpyDataset
+from src.helpers.recreate_preprocessor import recreate_preprocessor
+
+
+def predict_onnx(model, dataset: JaqpotpyDataset, request):
+    sess = InferenceSession(model.SerializeToString())
+    onnx_prediction = sess.run(
+        None, {"float_input": dataset.X.to_numpy().astype(np.float32)}
+    )
+    if len(request.model['dependentFeatures']) == 1:
+        onnx_prediction = onnx_prediction[0].reshape(-1, 1)
+
+    if request.model['extraConfig']['preprocessors']:
+        for i in reversed(range(len(request.model['extraConfig']['preprocessors']))):
```

Comment:
do we run them from end to finish? is the reversed the right way? I was thinking to run from 0 to array.length

Reply:
I want to discuss it in the meeting. Its a question that i have since yesterday. Right now, we do the opposite in jaqpotpy (i.e. what you suggested) but i am not sure that its correct